src/Components/Components/src/Routing/Router.cs

```diff
@@ -114,8 +127,21 @@ private static string StringUntilAny(string str, char[] chars)
                 : str.Substring(0, firstIndex);
         }
 
+        private void RefreshRouteTable()
+        {
+            var assemblies = AdditionalAssemblies == null ? new[] { AppAssembly } : new[] { AppAssembly }.Concat(AdditionalAssemblies);
+            var assembliesSet = new HashSet<Assembly>(assemblies);
+            if (_assemblies.Count != assembliesSet.Count)
```

Comment:
This seems a bit risky. With the API as-is, the developer could mutate `AdditionalAssemblies` arbitrarily, without necessarily changing the number of entries in it.

Rather than trying to detect modifications, perhaps it would be safer to have an explicit API the developer calls when they want to refresh the route table. This would also save us from going through this process on every navigation, even when not lazy-loading.

Reply:
I think it is fine to do this, no need to provide an explicit API for it. We can check the if the sizes are different and refresh (most common case) and if they are the same just check one set against the other for equality.

Reply:
> they are the same just check one set against the other for equality.

I assume you mean a deep equality check. I agree that would be OK since it's hard to think of a case where the number of entries should exceed (say) 100. When it gets much beyond that the perf implications would kick in on wasm.

Reply:
I was originally in favor of having an explicit callback that the user would invoke. Javier turned me around on this approach of refreshing it our selves because:

- It hides an implementation detail away from the user
- It avoids users landing in a pit of failure when they don't call the `RefreshRouteTable` method when they should

As for how the check should be done, I figured a length check should be enough since the changes to AdditionalAssemblies will likely be additive 99% of the time (as Javier mentioned above). We can add a secondary (deep) equality check to help us sleep better at night.